x-pack/plugins/ml/server/routes/trained_models.ts

```diff
@@ -237,6 +239,112 @@ export function trainedModelsRoutes({ router, routeGuard }: RouteInitialization)
       })
     );
 
+  /**
+   * @apiGroup TrainedModels
+   *
+   * @api {get} /internal/ml/trained_models/ingest_pipelines Get ingest pipelines
+   * @apiName GetIngestPipelines
+   * @apiDescription Retrieves pipelines
+   */
+  router.versioned
+    .get({
+      path: `${ML_INTERNAL_BASE_PATH}/trained_models/ingest_pipelines`,
+      access: 'internal',
+      options: {
+        tags: ['access:ml:canGetTrainedModels'], // TODO: update permissions
+      },
+    })
+    .addVersion(
+      {
+        version: '1',
+        validate: false,
+      },
+      routeGuard.fullLicenseAPIGuard(async ({ client, request, mlClient, response }) => {
+        try {
+          const body = await modelsProvider(client).getPipelines();
+          return response.ok({
+            body,
+          });
+        } catch (e) {
+          return response.customError(wrapError(e));
+        }
+      })
+    );
+
+  /**
+   * @apiGroup TrainedModels
+   *
+   * @api {post} /internal/ml/trained_models/simulate_pipeline simulate the ingest pipelines
+   * @apiName SimulateIngestPipeline
+   * @apiDescription Simulates the pipeline
+   */
+  router.versioned
+    .post({
+      path: `${ML_INTERNAL_BASE_PATH}/trained_models/simulate_pipeline`,
+      access: 'internal',
+      options: {
+        tags: ['access:ml:canCreateTrainedModels'],
+      },
+    })
+    .addVersion(
+      {
+        version: '1',
+        validate: {
+          request: {
+            body: simulateIngestPipelineSchema,
+          },
+        },
+      },
+      routeGuard.fullLicenseAPIGuard(async ({ client, request, mlClient, response }) => {
+        try {
+          const { docs, pipeline } = request.body;
+          const body = await modelsProvider(client).simulatePipeline(docs, pipeline);
+          return response.ok({
+            body,
+          });
+        } catch (e) {
+          return response.customError(wrapError(e));
+        }
+      })
+    );
+
+  /**
+   * @apiGroup TrainedModels
+   *
+   * @api {post} /internal/ml/trained_models/create_inference_pipeline creates the pipeline with inference processor
```

Comment:
This API needs to be added to the config for the docs script - https://github.com/elastic/kibana/blob/main/x-pack/plugins/ml/scripts/apidoc_scripts/apidoc_config/apidoc.json